synapse/handlers/user_directory.py

```diff
@@ -232,7 +228,7 @@ async def _handle_deltas(self, deltas: List[Dict[str, Any]]) -> None:
                         continue
 
                     if change:  # The user joined
-                        event = await self.store.get_event(event_id, allow_none=True)
```

Comment:
Right below this we assume that the event exists, so we either need to not `allow_none` *or* handle `event` being `None`.

Reply:
Hmm. So I guess this could break if we get a rejected event in our state delta? Since this is a background job that won't make any progress if it breaks(?) I'd be tempted to keep the `allow_none` and ignore if it is `None` (as if change was false).

Reply:
Ah, that probably makes more sense. 👍

Reply:
Looking closer at this code, I don't think just ignoring it is the right change, I think we still need to handle that it occurred, but not try to do anything with profile data?

Reply:
Oh, possibly. TBH I'm not too concerned as this is a) for the user directory and b) for a case which really shouldn't happen (though has in the past)

Reply:
I made this change!